backend/src/auth/services/auth_service.py

```diff
@@ -1,82 +1,225 @@
+        """
+        # Publish user created event to the cache handler
+        publish_handler_event(
+            pattern="user_created",
+            data={
+                "id": user_id,
+                "email": user.email,
+                "username": user.username,
+                "profile": {
+                    "name": profile.name,
+                    "surname": profile.surname,
+                    "image_url": profile.image_url,
+                },
+            },
+        )
+
+    def register(self, user: RegisterUserModel, db: Session) -> Optional[JSONResponse]:
+        """
+        Registers a new user and returns a JSONResponse.
+
+        Args:
+            user (RegisterUserModel): The user to register.
+            db (Session): The database session.
+
+        Returns:
+            Optional[JSONResponse]: The JSONResponse with the registration status.
+        """
+        # Check if the username is already taken
         if self.repository.get_by_username(user.username, db):
             raise UsernameTaken
+        # Check if the email is already taken
         elif self.repository.get_by_email(user.email, db):
             raise EmailTaken
+
         try:
-            self.repository.create_user(user, db)
-            return RegisterResponse(
-                message="User registered successfully",
+            # Create the user in the database
+            user_db = self.create_user(user=user, db=db)
+            # Check if the user creation was successful
+            if not user_db:
+                raise InvalidUserData
+
+            # Publish user created event
+            self.publish_user_created_event(
+                user_id=str(user_db.id), user=user, profile=user.profile
+            )
+
+            # Return successful registration message
+            return JSONResponse(
+                content=RegisterEndpoint(
+                    message="User registered successfully"
+                ).model_dump(),
+                status_code=201,
             )
         except Exception:
-            raise DetailedHTTPException
+            raise RegisterFailed
 
-    def login(self, user: GetUser, db) -> Optional[LoginResponse]:
+    def login(self, user: LoginUserModel, db: Session) -> Optional[JSONResponse]:
+        """
+        Logs in the user and returns a JSONResponse with authentication token.
+
+        Args:
+        - user: LoginUserModel object containing username and password
+        - db: database session
+
+        Returns:
+        - Optional[JSONResponse]: JSONResponse object with authentication token
+        """
+        # Authenticate the user
         auth_user = self.authenticate_user(
             username=user.username,
             password=user.password,
             db=db,
         )
         if not auth_user:
             raise NotAuthenticated
+
+        # Set user as active
         self.repository.set_is_active(auth_user, db)
```

Comment:
What is the purpose of it? If we want to check if account is active we should do this in register endpoint instead of login.

Reply:
and it's not like this:
a user registers -> data is sent to the database, but he is not active yet because he has not logged in.
And during the login process, we first check whether such a user exists and is already registered in the database,
if so, only then do we give him the opportunity to change the flag to active user because he logs in and give him a token to log in?

Reply:
this check this user is auth in db and if is, we change him flag active and token to log in.